src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1684,6 +1685,22 @@ bool esbmc_parseoptionst::process_goto_program(
       return true;
     }
 
+    if(cmdline.isset("output-goto"))
+    {
+      msg.status("Writing GOTO program to file");
+      std::string fullname = cmdline.args[0];
+      // ! assume correct filename
+      size_t lastindex = fullname.find_last_of(".");
+      std::string gbfname = fullname.substr(0, lastindex) + ".goto";
+      std::ofstream oss(gbfname, std::ios::binary);
+      if(write_goto_binary(oss, context, goto_functions))
```

Comment:
Might be better to print the file output as well. `msg.status(fmt::format("Writing GOTO program into {}", gbfname));`